### PR TITLE
test: connectFactoryObservable updates, subscriptions and errors.

### PR DESCRIPTION
**Pull request by voliva**
These covers the 3 remaining points identified for connectFactoryObservable in #20

### CODE REVIEW

File: test/connectFactoryObservable.test.tsx [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 
 const wait = (ms: number) => new Promise(res => setTimeout(res, ms))
 
 describe("connectFactoryObservable", () => {
   const originalError = console.error
   beforeAll(() => {
     console.error = (...args: any) => {
-      if (/Consider adding an error/.test(args[0])) {
+      if (
+        /Uncaught 'controlled error'/.test(args[0]) ||
```
Comment (voliva) [inline test/connectFactoryObservable.test.tsx:16]: That's interesting - React stills throws the error (so it shows up as an uncaught exception in the console) even if there's an ErrorBoundary. Not only that, but CRA will overlay a ugly error message on the whole body if it's in dev mode, telling you that you should fix that error that happened.

It only uses the ErrorBoundary to handle that specific component crashing, and in prod it doesn't show the overlayed message.

File: test/connectFactoryObservable.test.tsx [hunk 2]
```diff
@@ -20,9 +27,43 @@ describe("connectFactoryObservable", () => {
   })
   describe("hook", () => {
     it("returns the latest emitted value", async () => {
-      const [useNumber] = connectFactoryObservable((id: number) => of(id))
+      const valueStream = new BehaviorSubject(1)
```
Comment (bhavesh-desai-scratch) [inline test/connectFactoryObservable.test.tsx:30]: You could simplify this by removing the ID from the output. It is not necessary for this test.

File: test/connectFactoryObservable.test.tsx [hunk 3]
```diff
@@ -20,9 +27,43 @@ describe("connectFactoryObservable", () => {
...
-      expect(result.current).toBe(1)
+      expect(result.current).toBe("1:1")
+
+      actHook(() => {
+        valueStream.next(3)
+      })
+      expect(result.current).toBe("1:3")
+    })
+
+    it("shares the subscription among all of the subscribers of the same key", async () => {
```
Comment (bhavesh-desai-scratch) [inline test/connectFactoryObservable.test.tsx:43]: Is the key not an implementation detail of `createFactoryObservable`?

Should this not be something like:

Multiple components using a hook returned from `createFactoryObservable`  with the same parameters share a multicasted subscription?

Comment (voliva) [reply test/connectFactoryObservable.test.tsx:43]: Yep, that's a better test description 👍

File: test/connectFactoryObservable.test.tsx (hunk 2 above)
Comment (voliva) [reply test/connectFactoryObservable.test.tsx:30]: Good catch!

**Pull request merged**